packages-ui/roosterjs-react/lib/emoji/plugin/createEmojiPlugin.ts

```diff
@@ -0,0 +1,426 @@
+import * as ReactDOM from 'react-dom';
+import EmojiPane, { EmojiPaneMode } from '../components/EmojiPane';
+import showEmojiCallout from '../components/showEmojiCallout';
+import { AriaAttributes } from '../type/AriaAttributes';
+import { Editor } from 'roosterjs-editor-core';
+import { Emoji } from '../type/Emoji';
+import { KeyCodes } from '@fluentui/react/lib/Utilities';
+import { matchShortcut } from '../utils/searchEmojis';
+import { MoreEmoji } from '../utils/emojiList';
+import { ReactEditorPlugin, UIUtilities } from '../../common/index';
+import { replaceWithNode } from 'roosterjs-editor-api';
+import { Strings } from '../type/Strings';
+import {
+    ChangeSource,
+    PluginDomEvent,
+    PluginEvent,
+    PluginEventType,
+    PositionType,
+} from 'roosterjs-editor-types';
+import {
+    EmojiDescriptionStrings,
+    EmojiFamilyStrings,
+    EmojiKeywordStrings,
+} from '../type/EmojiStrings';
+
+const EMOJI_SEARCH_DELAY = 300;
+const INTERNAL_EMOJI_FONT_NAME = 'EmojiFont';
+const EMOJI_FONT_LIST =
+    "'Apple Color Emoji','Segoe UI Emoji', NotoColorEmoji,'Segoe UI Symbol','Android Emoji',EmojiSymbols";
+const KEYCODE_COLON = 186;
+const KEYCODE_COLON_FIREFOX = 59;
+
+// Regex looks for an emoji right before the : to allow contextual search immediately following an emoji
+// MATCHES: 0: 😃:r
+//          1: 😃
+//          2: :r
+
+const EMOJI_BEFORE_COLON_REGEX = /([\u0023-\u0039][\u20e3]|[\ud800-\udbff][\udc00-\udfff]|[\u00a9-\u00ae]|[\u2122-\u3299])*([:;][^:]*)/;
+
+/**
+ * Interface to customize the emoji plugin
+ */
+export interface EmojiPluginOptions {
+    searchPlaceholder?: string;
+    searchInputAriaLabel?: string;
```

Comment:
are these for localized string? If so, please follow the existing pattern for localized strings in other components/plugins. And I'm ok if we do not expose these options for now and add it later.